consensus/src/network_interface.rs

```diff
@@ -119,7 +122,7 @@ impl ConsensusNetworkSender {
         recipients: impl Iterator<Item = PeerId>,
         message: ConsensusMsg,
     ) -> Result<(), NetworkError> {
-        let protocol = ProtocolId::ConsensusDirectSend;
+        let protocol = ProtocolId::ConsensusDirectSendCbor;
```

Comment:
Did you test communication between a node that has the Cbor, and one that doesn't?  As far as I see, it should handle receiving old messages, but it will also send Cbor messages to nodes that might not support it.  It should be able to tell which protocols are supported in the handshake, and send it accordingly.

Reply:
`network_endpoint_config` is supposed to address this, I appends the ConsensusDirectSendCbor to the accepted protocol vec so it should receive messages from both.

Reply:
I guess there's a period when less than 2f+1 supports both protocols, need to use the old one until majority upgrades the software and switch the config

Reply:
Yes, you support in `network_endpoint_config` the new software.  But, if I don't have the change on my node (old software), you send me CBOR messages, and I just drop them as errors.

I just want to make sure we're in the mentality of backwards compatibility so that there's less downtime and it'll continue to run.

Reply:
I was hoping that the send api could return error that the protocol is not supported for this peer so that we can retry with another protocol but seems it's not the case

Reply:
Yeah, we should be able to do it without roundtripping the error, but it looks like the framework isn't there for it.

What we should have is to know the supported protocols of each peer, and to appropriately provide possible protocols to be sent on the NetworkSender.

Reply:
It's all yours :D

Reply:
Can we add tests for the expected behavior around nodes with different protocols available?

Reply:
I suppose we have existing tests for different network protocols given we have dozens of protocols already